doc/reference/gerbil/core/control-flow.md

```diff
@@ -430,63 +430,248 @@ For simple, one-off conditional execution, `if`, `when`, or `cond` are often mor
+Evaluates a sequence of expressions in order, but returns the value of the **first** expression.
+
+The `begin0` macro first evaluates `expr` and saves its value. It then evaluates all subsequent `rest ...` expressions in order for their side effects, **ignoring their return values**. Finally, it returns the saved value of the initial `expr`.
+
+This is in contrast to the standard `begin` form, which returns the value of the *last* expression.
+
+
+::: tip Example
+Retrieve a user's data and then evicts it from cache.
+
+```scheme
+> (def (get-and-evict! cache key)
+    (begin0
+      (hash-ref cache key)
+      (hash-remove! cache key)))
+
+> (def user-cache (hash ("user:1" "Alice") ("user:2" "Bob")))
+
+> (get-and-evict! user-cache "user:1")
+"Alice"
 
-## @list
+;; After the call, the cache no longer contains Alice's entry.
+> (hash-key? user-cache "user:1")
+#f
 ```
-\[<list-expression-body> ...\]
-(@list <list-expression-body> ...)
+:::
+
+### Context and Usage
+
+Use `begin0` when you need to evaluate a series of expressions for their side effects but require the result of the very first expression. It elegantly combines retrieving a value and then performing cleanup or mutation operations related to that value.
+
+### See Also
 
-list-expression-body ...:
- . '<s-expression>
- . `<s-expression>
- . tail
- :: <s-expr>
- <s-expr> \... <list-expression-body> ...
- <s-expr> <list-expression-body> ...
+`begin`
+
+## @list <Badge type="tip" text="Gerbil" vertical="middle" />
+
+```scheme
+[<list-item> ...]
+(@list <list-item> ...)
+
+list-item:
+  <list-expression> "..."
+  . <tail-expression>
+  :: <tail-expression>
+  <expression>
 ```
 
-The list constructor expression, normally implied with the `[]` reader
-macro.
+The primary list constructor, used with the `[...]` reader macro, which supports in-place list splicing and dotted-tail notation.
+
+`@list` (and its reader macro `[...]`) provides a flexible way to construct lists. It evaluates its arguments and combines them into a new list. In almost all cases, you will use the convenient `[...]` syntax.
 
-## delay
+Unlike the standard `list` procedure, the `[...]` constructor gives special meaning to certain symbols:
+
+* The ellipsis `...` has a conditional behavior based on the expression that precedes it:
+    * If the preceding expression evaluates to a list, it acts as a **splicing operator**, unpacking the elements of that list into the new list.
+    * If the preceding expression is *not* a list, it **discards** the value of that expression from the final result.
+
+* Both the dot `.` and colon-colon `::` act as **dotted-tail constructors**. They are functionally equivalent and make the *following expression* the final `cdr` (tail) of the list. They must be the penultimate item in the expression.
+
+There is also a special unwrapping rule: if a list construction with a splicing `...` results in a single-element list, the result is the element itself (e.g., `[42 ...] => 42`).
```

Comment:
which is inconsistent with the handling of non list/improper list values above.

I think this is a bug and we should fix it (and update the docs once it is fixed).